lib/unison-prelude/src/Unison/Debug.hs

```diff
@@ -159,8 +158,7 @@ debug flag msg a =
 debugM :: (Show a, Monad m) => DebugFlag -> String -> a -> m ()
 debugM flag msg a =
   whenDebug flag do
-    pTraceM (msg <> ":\n")
-    pTraceShowM a
+    traceM (msg <> ":\n" <> into @String (pShow a))
```

Comment:
Having two trace statements would sometimes cause interleaving; combining them into one is much more reliable.